src/main/java/net/mcreator/element/types/Achievement.java

```diff
@@ -96,4 +102,23 @@ public boolean hasRewards() {
 		};
 	}
 
+	@Override public Collection<String> getUsedElementNames() {
+		List<String> elements = new ArrayList<>();
+		elements.add("CUSTOM:" + rewardFunction);

```

Comment:
Reward function can be null or empty, which will add entry "CUSTOM:" or "CUSTOM:null" which is not correct. Check if there are more cases of this and correct.

Reply:
Attempt 4

Reply:
✔️ Fixed unless the same should be applied to `getUsedElementMappings()`.

Reply:
All methods should only return valid names and entries that are actually used